### PR TITLE
288 product detail page

### LINKED ISSUE #288: [Product] Product/Product Family detail page - part 1
- [x] Add template code and metadata info for template
- [ ] Coveo data loading via template and store it in local (session or longer time)
- [x] Product Hero (details, image viewer, ...)

### PR DESCRIPTION
**Pull request by amlan18041996**
Fix #288 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/
- After: https://288-product-detail-page--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/mica
- After: https://288-product-detail-page--danaher-ls-aem--hlxsites.hlx.page/us/en/products/family/imagexpress-pico-automated-cell-imaging-systems

### CODE REVIEW

**Review by davenichols-DHLS — APPROVED** [review]
Nice!  Thanks Amlan, I am checking the products and can see the data coming through, this is exciting.

**Review by rgravitvl — APPROVED** [review]

File: templates/productDetail/productDetail.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      host: 'stage.lifesciences.danaher.com',
+      internal: false,
+    },
+    q: `@${qParam}==${sku}`,
+    pipeline: 'Product Details',
+  };
+  return payload;
+}
+
+async function makeCoveoApiRequest(path, accessParam, payload = {}) {
```
Comment (duynguyen) [inline templates/productDetail/productDetail.js:19]: is this a copy of the same function from header block? if mostly the same, let's try to move the function to scripts.js and reuse it in both here and header.js

File: templates/productDetail/productDetail.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    if (currentActive) currentActive.classList.toggle('active');
+    if (e.target.src !== selectedImage.src) {
+      selectedImage.src = e.target.src;
+      e.target.classList.toggle('active');
+    }
+  }
+}
+
+function loadMore() {
+  const allImageContainer = document.querySelector('.vertical-gallery-container div div:has(img.active)');
```
Comment (duynguyen) [inline templates/productDetail/productDetail.js:51]: as already seen before in carousel block, `:has` is not supported in older versions of Firefox and might break for many users.

Comment (davenichols-DHLS) [reply templates/productDetail/productDetail.js:51]: @amlan18041996  - would you mind please updating as per @duynguyen 's feedback?

**Review by duynguyen — APPROVED** [review]

**Pull request merged**